articles/qos_deadline_liveliness_lifespan.md

```diff
@@ -25,102 +25,155 @@ Glossary:
+They could fail with an exception that notes that an unsupported QoS policy was requested or they can choose to continue operating with a reduced QoS policy.
+In the case where a vendor chooses to continue operating with a reduced policy it must trigger a status event for the resource that can be handled by the new callback outlined above.
+This provides an application a way of being notified that even though their resource is operating, it is doing so with a reduced QoS.
 
-- How do Deadlines and Liveliness impact Services and Actions?
-  - Actions and Services are slightly different in that they are composed of multiple Topics. It may make sense for a Service to specify a Liveliness policy on its own DataWriter, but it likely doesn't need the same policy on the DataReader.
-  - One option would be to only apply the QoS settings to one or another of the topics. So Liveliness may only apply to the DataWriters in Services and Actions and the Deadlines might apply only to the "update" topic data writer for Actions.
-  - Another option would be to modify the interface to allow for different QoS policies for the different underlying Topics, though this adds more complexity.
-- How do these new QoS settings impact non-DDS based rmw implementations?
-  - While these QoS policies are native to the DDS standard, any non-DDS based rmw layer will not have native support for these policies. What does this mean for those implementations? Will they be forced to implement these features in their rmw_implementation layer to be officially supported?
+## FAQ
 
+- How does the Deadline policy take into account the additional overhead of ROS (such as deserialization) when determining if a deadline was missed?
+  - As a simplification it is not going to attempt to take into account any ROS overhead. A deadline will be considered missed if the rmw layer does not receive a message by the deadline and not if the user application on top of ROS does not receive it by the deadline. A new deadline policy could be added later that takes this into account.
+- What happens if multiple policy violations occur between the invocation of the callback? 
+  - This is dependent on the rmw implementation. If the rmw layer supports tracking multiple status events then it can return them in subsequent calls to rmw_take_status. If it does not support tracking multiple events at once then it may just return one of the events, such as the first or the last to occur in the sequence.
+- How do these QoS policies impact Actions?
+  - The existing Actions interface exposes that Actions are implemented with multiple Topics and it exposes the QoS settings for those topics. It is therefore up to the application implementing the action to specify the QoS policies for the different Topics. A new interface will need to be added to allow Actions to specify a ResourceStatusHandler, but that will be added in a later design review.
```

Comment:
Taking a completely different approach for actions than for topics and services seems odd to me. Why can't we treat them in the same way?

Reply:
As far as I know, they currently are different in that Actions aren't supported natively in the rmw interface like Topics and Services. Additionally, their design and api expose them as being implemented on top of topics/services, so it makes it more challenging to abstract the QoS settings treating Actions as a black box when current users can already change behaviors via the exposed QoS settings for the underling Topics/Actions. 

If it seems like I'm drawing an arbitrary line in the sand I'd say it's because I was focused on the "rcl" native concepts for this design and putting off "rcl_action", which is built on top of that, in order to try to keep the scope of the design as small as possible for the first iteration.

Reply:
> Additionally, their design and api expose them as being implemented on top of topics/services

This is a problem itself. I'm sure @mkhansen-intel would have Words To Say on this topic.

Reply:
I think we should include actions in the design, because just like rclcpp being backed by rcl and rmw is an implementation detail, actions being implemented on topics and services is also an implementation detail, and so they shouldn't left out just because they're not in the rmw API (by the way, being in or out of rcl proper doesn't imply anything to me, being in or out of rmw does).

On the other hand, I don't see why it being obvious that actions are built on top of topics and services is a problem, i.e. having accessors for the underlying services and pub/subs. It doesn't at all imply first-class versus second-class, just derived versus primitive. What matters are cases like this where we could leave actions out of the design or include them, in which case I say include them, even if it's just to say "we will explicitly not have this QoS for actions due to `<reason>`".

Reply:
I think treating actions like first-class citizens means that they are equal in the design stage. If the design doesn't include actions, it's incomplete. If it's going to be incomplete, at least it should specify that the design for actions will be forthcoming. However, in this case, I really think thought should be put into how to make using actions as similar to topics and services as possible. Maybe an array of QOS structs is passed into the Actions constructor to specify the policies for all of the topics and services contained within.  Maybe I missed it but I don't think that's what this is proposing.